src/troubleshooting.rst

```diff
@@ -0,0 +1,204 @@
+``job-activity.log``
+   This file records the interaction Cylc has had with a job, namely submission
+   and polling. This can be useful in determining the cause of job submission
+   failures.
+
+   You can view this in the GUI or on the command line using
+   ``cylc cat-log <workflow>//<cycle>/<task>/<job> -f a``.
+
+
+Problems
+--------
+
+
+Job Status Isn't Updating
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Cylc keeps track of a job's progress in one of two ways (according to how
+the platform the job was submitted to is configured):
+
+* Jobs send messages to the scheduler (push).
+* The scheduler polls jobs (pull).
+
+In either case, the job will also write its updates to the ``job.status`` file.
+
+This is what the ``job.status`` file should look like for a successful job,
+note the ``SUCCEEDED`` line:
+
+.. code-block::
+
+   CYLC_JOB_RUNNER_NAME=background
+   CYLC_JOB_ID=12345
+   CYLC_JOB_RUNNER_SUBMIT_TIME=2000-01-01T00:00:00
+   CYLC_JOB_PID=108179
+   CYLC_JOB_INIT_TIME=2000-01-01T00:10:00
+   CYLC_JOB_EXIT=SUCCEEDED
+   CYLC_JOB_EXIT_TIME=2000-01-01T01:30:00
+
+If the ``job.status`` file is showing something different to what the GUI or
+Tui is showing, then...
+
+.. rubric:: If your platform uses push communication:
+
+If messages aren't getting back to the scheduler, there should be some
+evidence of this in the ``job.err`` file, likely either an error or a
+traceback.
+
+Likely causes:
+
+* There is a network issue.
+* TCP ports are not open (zmq communications).
+* Non-interactive SSH has not been correctly configured (ssh communications).
+
+.. rubric:: If your platform uses pull communication:
+
+Firstly, check the polling interval, it's possible that the scheduler has been
+configured to poll infrequently and you need to wait for the next poll, or use
+the ``cylc poll`` command (also available in the GUI).
+
+Use the ``cylc config`` command to inspect the platform's configuration to
+determine the configured polling schedule.
+
+Then check the ``job-activity.log`` file, there may have been a problem polling
+the remote platform, e.g. a network or configuration error.
+
+Likely causes:
+
+* The platform is down (e.g. all login nodes are offline).
+* There is a network issue.
+* Non-interactive SSH has not been correctly configured.
+
+
+My Job Submit-Failed
+^^^^^^^^^^^^^^^^^^^^
+
+A submit-failed job means one of three things:
+
+1. There is a Bash syntax error in the task configuration.
+
+   E.G. the following ``script`` has a syntax error, it is missing a
+   ``"`` character:
+   
+   .. code-block:: cylc
+   
+      [runtime]
+          [[foo]]
+              script = echo "Hello $WORLD
+   
+   This will result in a submission-failure which should appear in the
+   ``job-activity.log`` file (and also the scheduler log) something like this:
+   
+   .. code-block::
+   
+      /path/to/job.tmp: line 46: unexpected EOF while looking for matching `"'
+      /path/to/job.tmp: line 50: syntax error: unexpected end of file
```

Comment:
```suggestion
      [jobs-submit cmd] (prepare job file)
      [jobs-submit ret_code] 1
      [jobs-submit err]
      /path/to/job.tmp: line 46: unexpected EOF while looking for matching `"'
      /path/to/job.tmp: line 50: syntax error: unexpected end of file
      
Cylc checks job syntax just before submission (because active broadcasts may alter task scripting on the fly) and aborts the process if errors are found.
```
I think it's worth adding more context here - it shows that job preparation failed, not execution - which might be expected for a bash syntax error.